File: tests/fixtures/java/src/main/java/com/example/Gamma.java
package com.example;

import static com.example.alpha.Alpha.helper;
import static com.example.beta.Beta.helper;

/**
 * Case c (D3): project-wide name collision.
 *
 * Both static imports bring a `helper` method into this file's unqualified
 * scope. javac would reject this outright ("reference to helper is
 * ambiguous") — but it is syntactically valid Java, and codegraph's
 * resolver, which does no compile-time static-import-collision checking,
 * must independently report AMBIGUOUS rather than blending (D3's exact
 * failure mode) or silently picking one.
 */
public class Gamma {
    public static void dispatch() {
        helper();
    }
}

File: tests/fixtures/c-cpp/gamma.cpp
#include "gamma.h"

#include "alpha.h"
#include "beta.h"

using namespace alpha;
using namespace beta;

// Case c (D3): project-wide name collision.
//
// Both using-directives make `helper` visible for unqualified lookup; a
// real compiler only errors at the point of actual ambiguous use ("call of
// overloaded 'helper()' is ambiguous"), not at the using-directives above —
// so this is valid to write, and exactly the kind of ambiguity codegraph's
// resolver (no compile-time overload resolution; design principle 4, "not a
// compiler") must independently detect rather than blend (D3's exact
// failure mode) or silently pick one.
void dispatch() {
    helper();
}
